### ISSUE TITLE
Train model

### ISSUE THREAD

**Issue by sowmyay** [sowmyay/w4c#5]
Train the first model: UNet + pre-trained ResNet50 encoders

Ensure that:

- [ ] You are saving the best checkpoint
- [ ] Monitoring training loss and validation loss on tensorboard
- [ ] Ensure optimum usage of GPUs by adjusting batch size, num workers, etc
